projects/charger/src/charger_controller.c

```diff
@@ -0,0 +1,119 @@
+#include "charger_controller.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "can_transmit.h"
+#include "charger_defs.h"
+#include "charger_events.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "generic_can_mcp2515.h"
+#include "mcp2515.h"
+#include "soft_timer.h"
+
+typedef enum { TX_CONTROL_START_CHARGING = 0, TX_CONTROL_STOP_CHARGING } TxControl;
+
+static SoftTimerId s_charger_timer_id;
+static ChargerVC s_vc = {
+  .values.complete.current = 0,                         //
+  .values.complete.voltage = CHARGER_BATTERY_THRESHOLD  //
+};
+static GenericCanMcp2515 s_generic_can;
+
+static Mcp2515Settings mcp2515_settings = {
+  .spi_port = SPI_PORT_2,                        //
+  .spi_baudrate = 6000000,                       //
+  .mosi = { .port = GPIO_PORT_B, .pin = 15 },    //
+  .miso = { .port = GPIO_PORT_B, .pin = 14 },    //
+  .sclk = { .port = GPIO_PORT_B, .pin = 13 },    //
+  .cs = { .port = GPIO_PORT_B, .pin = 12 },      //
+  .int_pin = { .port = GPIO_PORT_A, .pin = 8 },  //
+  .can_bitrate = MCP2515_BITRATE_250KBPS,        //
+  .loopback = false,                             //
+};
+
+static TxMsgData prv_build_charger_tx(TxControl charge) {
+  TxMsgData tx_data = {
+    .fields.max_voltage_high = s_vc.values.bytes.voltage_high,  //
+    .fields.max_voltage_low = s_vc.values.bytes.voltage_low,    //
+    .fields.max_current_high = s_vc.values.bytes.current_high,  //
+    .fields.max_current_low = s_vc.values.bytes.current_low,    //
+    .fields.control = charge,                                   //
+  };
+  return tx_data;
+}
+
+static void prv_periodic_charger_tx(SoftTimerId timer_id, void *context) {
+  TxMsgData tx_data = prv_build_charger_tx(true);
```

Comment:
Looks like you turned the `prv_build_charger_tx` argument into an enum, but forgot to use the enum values when calling it!

Reply:
Good catch whoops, updated calls to `prv_build_charger` to use a `TxControl` value as an argument